### PR TITLE
Add support for default datacenter parameter

### PR DESCRIPTION
**Pull request by miklinux**
This an initial PR is to add support for the default datacenter parameter.
The parameter can be set on initialization and overridden when needed.

I guess I also need to add some test case here, but need some advice here.

### CODE REVIEW

File: lib/diplomat/rest_client.rb [hunk 1]
```diff
@@ -219,6 +219,7 @@ def use_consistency(options)
       options[:consistency] ? [options[:consistency].to_s] : []
     end
 
+
```
Comment (pierresouchay) [inline lib/diplomat/rest_client.rb:222]: remove this extra-line (and thus rubocop won't complain)

File: lib/diplomat/rest_client.rb [hunk 2]
```diff
@@ -331,5 +332,6 @@ def valid_value_transactions
         verb if requires.include? 'Value'
       end
     end
+
```
Comment (pierresouchay) [inline lib/diplomat/rest_client.rb:335]: Remove this extra-line

File: lib/diplomat/configuration.rb [hunk 3]
```diff
@@ -18,11 +18,13 @@ def self.parse_consul_addr
     # Override defaults for configuration
     # @param url [String] consul's connection URL
     # @param acl_token [String] a connection token used when making requests to consul
+    # @param dc [String] consul's datacenter to use as default
```
Comment (pierresouchay) [inline lib/diplomat/configuration.rb:21]: Please specify somewhere that if nil -> fallback on agent's default DC and specify that using DC in any endpoint overrides this setting

**Review by pierresouchay — CHANGES_REQUESTED** [review]
Hello @miklinux,

Thanks a lot for your contribution. Appart from some complaints from our friend Rubocop, it looks very nice.

About tests, I would recommend adding just a few tests to ensure precedence rules are respected (aka: what happens dc is specific in the config and also in a request).

Thanks a lot for your contribution!

**Comment by miklinux** [comment]
Hi @pierresouchay,
I've just pushed the requested changes for you to review.

**Comment by pierresouchay** [comment]
@miklinux can you resolve the conflicts, so we can merge it?

**Pull request closed without merging**